### PR TITLE
C++20: use view() instead of str() to avoid a copy, use starts_with

### PR DESCRIPTION
**Pull request by servusdei2018**
## Description
Use `view()` instead of `str()`, `starts_with()` instead of `_Starts_with()`.

### Why
Resolves two TODO items

### What

One instance of a `str()` invocation was replaced with `view()`, avoiding an unnecessary copy. Additionally, the private function `_Starts_with()` was replaced with C++20's `starts_with()`.

## Changelog
Should this change be included in the release notes: _no_

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/react-native-windows/pull/13218)

### CODE REVIEW

**Comment by marlenecota** [comment]
@servusdei2018 - please run `yarn format` and push those changes :)

**Comment by servusdei2018** [comment]
Hm I tried running `yarn format` but git doesn't register any changes:

```sh
react-native-windows$ yarn format
yarn run v1.22.22
$ format-files -i -style=file


ran clang-format on 1119 files
Done in 3.35s.
react-native-windows$ git status
On branch servusdei2018-patch-1
Your branch is up to date with 'origin/servusdei2018-patch-1'.

nothing to commit, working tree clean
```

File: change/react-native-windows-f53a2b65-d364-49a5-978c-5d64e08c22ea.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "none",
```
Comment (jonthysell) [inline change/react-native-windows-f53a2b65-d364-49a5-978c-5d64e08c22ea.json:2]:
```suggestion
  "type": "prerelease",
```

File: change/react-native-windows-f53a2b65-d364-49a5-978c-5d64e08c22ea.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+  "comment": "Use view() instead of str(), starts_with() instead of _Starts_with()",
+  "packageName": "react-native-windows",
+  "email": "nate@bracy.dev",
+  "dependentChangeType": "none"
```
Comment (jonthysell) [inline change/react-native-windows-f53a2b65-d364-49a5-978c-5d64e08c22ea.json:6]:
```suggestion
  "dependentChangeType": "patch"
```

File: package.json [hunk 3]
```diff
@@ -65,5 +65,6 @@
   },
   "resolutions.justification": {
     "z-schema": "CVE-2021-3765 in validator. z-schema is used by rush which is a dependency of lage so should not be executed in this repo"
-  }
```
Comment (jonthysell) [inline package.json:68]: Your change shouldn't need to modify the package.json, please revert.

**Review by jonthysell — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (servusdei2018) [reply package.json]: Reverted, thank you for the review!

**Review by marlenecota — APPROVED** [review]

**Pull request merged**